test/network.bats

```diff
@@ -171,11 +172,11 @@ function teardown() {
 
 	# make conmon non-executable to cause the sandbox setup to fail after
 	# networking has been configured
-	chmod 0644 /go/src/github.com/kubernetes-incubator/cri-o/conmon/conmon
+	chmod 0644 $CONMON_BINARY
 	run crioctl pod run --config "$TESTDATA"/sandbox_config.json
+	chmod 0755 $CONMON_BINARY
```

Comment:
I'm still learning the intricacies of .bats test files, but I think you want this chmod line after the echo and status lines.

Reply:
That's how it was originally. I moved it up as a safety against failing. Now that `teardown()` restores permissions I'm more amenable to moving it back down. But first: is your suggestion based on the use of `$output` and `$status`? If so, note that it's the BATS `run` command that sets those, and the `chmod` is not invoked via `run`. As I understand BATS, and as shown in my testing, this is safe. (Extra proof: if this was not the case, the `$status` check below would be testing the exit code of the `echo`). 

With all that said: my inclination is still to run `chmod` as early as possible, simply as a restore-the-system-ASAP practice. But I'm no BATS expert and, as always, there may be something I've missed.

Reply:
Still earning my wings with bats.  After poking further, I believe you're right about the run creating the status that's checked, not the chmod.  But I think it's cleaner to do it after, especially if someone is reading the code who's a bats newby like myself.  

Not a biggy either way.  When you asked about the teardown earlier the teardown function name was hidden by github on me and I hadn't realized you'd it up there already.  Definitely it's needed there.  I think it's ok to remove in the test function itself, but it certainly won't hurt to have it as you do.

Reply:
> But I think it's cleaner to do it after, especially if someone is reading the code who's a bats newby like myself.

...whereas someone even minimally familiar with Bats would see the chmod after the `[ $status ...` check, and scream "whoa", realizing that if the test fails, the system is left in a corrupt state. Then they have to go look for a `teardown`.

> I think it's ok to remove in the test function itself

That's something I can't do, and I think it may be the source of our difference in opinion: I think that when I destabilize a system, I should revert it back as quickly as possible, and as close as possible to the source of the harm. I should not rely on a faraway cleanup handler. (It's good to have one as backup, but only as backup). Does that explain my thinking?

Reply:
I would prefer as @edsantiago has it in this PR. teardown is a like a safety net. Best is to restore what we are changing as soon as possible.

Reply:
👍 Thanks for the insights Ed, that's fine by me.